### PR TITLE
OBGM-416 Unable to use product filter on add items to PO shipment

### PR DESCRIPTION
**Pull request by alannadolny**

### CODE REVIEW

File: grails-app/services/org/pih/warehouse/order/OrderService.groovy [hunk 1]
```diff
@@ -1013,28 +1015,27 @@ class OrderService {
 
     def getProductsInOrders(String[] terms, Location destination, Location vendor) {
         return OrderItem.createCriteria().list {
+            createAlias('product', 'prod', JoinType.LEFT_OUTER_JOIN)
+            createAlias('product.synonyms', 'syn', JoinType.LEFT_OUTER_JOIN)
```
Comment (kchelstowski) [inline grails-app/services/org/pih/warehouse/order/OrderService.groovy:1019]: out of curosity it'd be worth also to check if here `prod.synonyms` would work

File: grails-app/services/org/pih/warehouse/order/OrderService.groovy [hunk 2]
```diff
@@ -1013,28 +1015,27 @@ class OrderService {
...
-                            ilike("productCode", term)
-                            ilike("description", "%" + term)
+            if (terms) {
+                terms.each { term ->
+                    term = term + "%"
+                    or {
+                        ilike("prod.name", "%" + term)
+                        and {
+                            ilike("syn.name", "%" + term)
+                                eq("syn.synonymTypeCode", SynonymTypeCode.DISPLAY_NAME)
```
Comment (kchelstowski) [inline grails-app/services/org/pih/warehouse/order/OrderService.groovy:1035]: could you fix the indent here, so it's not acting like it'd be nested inside `ilike`?

File: grails-app/services/org/pih/warehouse/order/OrderService.groovy (hunk 1 above)
Comment (alannadolny) [reply grails-app/services/org/pih/warehouse/order/OrderService.groovy]: Yup, it works

**Review by kchelstowski — APPROVED** [review]

File: grails-app/services/org/pih/warehouse/order/OrderService.groovy [hunk 3]
```diff
@@ -1013,28 +1015,27 @@ class OrderService {
 
     def getProductsInOrders(String[] terms, Location destination, Location vendor) {
         return OrderItem.createCriteria().list {
+            createAlias('product', 'prod', JoinType.LEFT_OUTER_JOIN)
```
Comment (awalkowiak) [inline grails-app/services/org/pih/warehouse/order/OrderService.groovy:1018]: Make these aliases a `product` and `synonym` instead of `prod` and `syn`

**Review by awalkowiak — CHANGES_REQUESTED** [review]

**Review by awalkowiak — APPROVED** [review]

**Pull request merged**